File: client/lib/shared/http/models/user_profile_model.dart
import 'package:client/shared/http/models/picture_model.dart';

class UserProfileModel {
  final int id;
  final String name;
  final String? email;
  final String? description;
  final PictureModel? picture;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> json = <String, dynamic>{};
    json['id'] = id;
    json['name'] = name;
    json['picture'] = picture;
    json['description'] = description;
    json['email'] = email;

    return json;
  }

  UserProfileModel.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        description = json['description'],
        email = json['email'],
        name = json['name'],
        picture = json['picture'] != null
            ? PictureModel.fromJson(json['picture'])
            : null;
}

File: server/src/interfaces/UserProfileModel.ts
import { PictureModel, pictureModelQuery } from "./PictureModel";

export interface UserProfileModel {
  id: number;
  picture?: PictureModel;
  name: string;
  email?: string;
  description?: string;
  createdAt: Date;
  updatedAt: Date;
}

export const userProfileModelQuery = {
  id: true,
  name: true,
  email: true,
  description: true,
  createdAt: true,
  updatedAt: true,
  picture: {
    select: pictureModelQuery,
  },
};

export function convertQueryToUserProfileModel(query: any): UserProfileModel {
  return <UserProfileModel>{
    id: query.id,
    picture: query.picture,
    name: query.name,
    email: query.email,
    description: query.description,
    createdAt: query.createdAt,
    updatedAt: query.updatedAt,
  };
}
